sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuCast.scala

```diff
@@ -777,17 +768,65 @@ object GpuCast extends Arm {
     }
   }
 
+  def castStringToDecimal(
+      input: ColumnView,
+      ansiEnabled: Boolean,
+      dt: DecimalType): ColumnVector = {
+    // 1. Sanitize strings to make sure all are floating points
+    // 2. Identify all fixed point values
+    // 3. Cast String to newDt (newDt = dt.scale + 1). Promote precision if needed. This step is
+    //    required so we can round up if needed in the final step
+    // 4. Now cast to newDt to dt (Decimal to Decimal)
+    def getInterimDecimalPromoteIfNeeded(dt: DecimalType): DecimalType = {
+      if (dt.scale + 1 > dt.precision) {
```

Comment:
If we are adding 1 to the scale we always have to add one to the precision. Or we are reducing the amount of digits above the decimal point that we can support.